ext/datasources/helium/helium.c

```diff
@@ -75,7 +84,7 @@ static int verbose = 0;					/* Verbose messages */
 		 * If we're set to a Helium error at the end of the day,\
 		 * switch to a generic WiredTiger error.		\
 		 */							\
-		if (ret < 0 && ret > -31,800)				\
+		if (ret < 0 && (ret > -31800 || ret < -31999))		\
```

Comment:
This is clearly a bug fix, but it took me a couple of reads to convince myself that it's correct.  How about this (which I think is equivalent):

```
/* If we get an error that is not a system error (indicated by a negative number), and outside the range reserved for WiredTiger errors, map to a generic WiredTiger error. */
if (ret < -31999 || (ret > -31800 && ret < 0))
        ...
```